qmp/socket.go

```diff
@@ -68,6 +68,31 @@ func NewSocketMonitor(network, addr string, timeout time.Duration) (*SocketMonit
 	return mon, nil
 }
 
+// Listen creates a new SocketMonitor by listening to the provided socket file or address.
+// An error is returned if unable to listen at the specified file path or port.
+//
+// Listen willl wait for a QEMU socket connection using a variety connection types:
+//	Listen("unix", "/var/lib/qemu/example.monitor")
+//	Listen("tcp", "0.0.0.0:4444")
+func Listen(network, addr string) (*SocketMonitor, error) {
+	listener, listenerErr := net.Listen(network, addr)
```

Comment:
I'd just go with `l, err :=` in a small scope function like this.

Reply:
✓